storage/innobase/xtrabackup/test/t/dictionary.sh

```diff
@@ -0,0 +1,306 @@
+ # Following commands in function can fail. Either success or failure is OK
+ # Also used by 'trap'. Hence we unset the 'e' flag which Exit immediately if a
+ # command exits with a non-zero status.
+ set +e
+ TRX_PID=$1
+ vlog "Kill the Long running uncommited transaction after the backup"
+ kill_query_pattern "INFO like '%SLEEP%'"
+
+ kill -SIGKILL $TRX_PID
+
+ vlog "Wait for killed process to exit"
+ wait $TRX_PID
+ set -e
+}
+
+function check_rollback_msg() {
+FILE=$1
+egrep -q "Rolling back trx with id [0-9]+, [0-9]+ rows to undo" $FILE || die "Expected rollback didn't happen during prepare"
+}
+
+function create_tablespace() {
+ mysql test << EOF
+ CREATE TABLESPACE ts1 ADD DATAFILE 'ts1.ibd';
+ CREATE TABLE t11(a INT) TABLESPACE=ts1;
+ CREATE TABLE t12(a INT, b INT) TABLESPACE=ts1;
+EOF
+}
+
+function create_tables() {
+  TABLE=$1
+  # optional parameter 2 for TABLESPACE
+  TABLESPACE=${2:-}
+  if [ ! -z ${TABLESPACE} ];
+  then
+    TABLESPACE="TABLESPACE=$TABLESPACE"
+  fi
+  mysql test << EOF
+    CREATE TABLE $TABLE (
+    id INT PRIMARY KEY,
+    val INT NOT NULL DEFAULT 0
+    ) $TABLESPACE ENGINE=InnoDB
+EOF
+
+  mysql -e "INSERT INTO $TABLE (id) VALUES (1),(2),(3),(4),(5),(6),(7),(8),(9),(10)" test
+  vlog "Long running transaction that modifies table data but is not committed at the time of backup"
+  $MYSQL $MYSQL_ARGS -e "\
+   BEGIN;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   INSERT INTO $TABLE (id) SELECT id + (SELECT MAX(id) FROM $TABLE) FROM $TABLE;\
+   SELECT SLEEP (10000);
+ " test &
+ uncommitted_trx=$!
+
+ trap "kill_bg_trx $uncommitted_trx" EXIT
+
+ wait_for_bg_trx
+
+ vlog "Backup"
+ rm -rf $topdir/backup
+ xtrabackup --backup --target-dir=$topdir/backup 2>&1 | tee $topdir/pxb.log
+
+ kill_bg_trx $uncommitted_trx
+
+ vlog "Record db state"
+ record_db_state test
+
+ vlog "Prepare the backup"
+ xtrabackup --prepare --target-dir=$topdir/backup 2>&1 | tee $topdir/pxb_prepare.log
+ check_rollback_msg $topdir/pxb_prepare.log
+
+ vlog "Restore"
+ stop_server
+ rm -rf $MYSQLD_DATADIR/*
+ xtrabackup --move-back --target-dir=${topdir}/backup
+ start_server
+ verify_db_state test
+}
+
+function create_part_table() {
+  vlog "Create partition table"
```

Comment:
Can you please create table with subpartition and do some uncommitted transaction. so we have rollbackup on partition where SDI is not stored

Reply:
subpartitions are not any different from the rollback perspective.